src/preprocess/base.py

```diff
@@ -43,7 +44,11 @@ def __init__(self, data_folder: Path = Path('data')) -> None:
             else:
                 folder_prefix = self.dataset
 
-            self.out_dir = self.preprocessed_folder / f'{folder_prefix}_preprocessed'
+            if self.analysis:
```

Comment:
wanted this functionality to remain with the base class to avoid writing the same function for the child class? does that make sense?

Reply:
yea, this makes sense to me